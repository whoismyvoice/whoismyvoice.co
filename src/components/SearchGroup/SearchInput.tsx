import * as React from 'react';

// Assets
import '../../styles/SearchInput.scss';

const isUsZip = /^(\d{5})(-\d{4})?$/;

interface Props {
  errorMessage?: string;
  name: string;
  pattern?: string;
  placeholder?: string;
}

interface State {
  isError: boolean;
}

export class SearchInput extends React.Component<Props, State> {
  static defaultProps = {
    errorMessage: undefined,
    pattern: '[0-9]*',
    placeholder: 'Enter Your Zip Code',
  };

  state = {
    isError: false,
  };

  constructor(props: Props) {
    super(props);
    this.state = {
      isError: props.errorMessage !== undefined,
    };
  }

  UNSAFE_componentWillReceiveProps(nextProps: Props): void {
    if (nextProps.errorMessage !== undefined) {
      this.setState({
        isError: true,
      });
    }
  }

  onChange(event: React.ChangeEvent<HTMLInputElement>): void {
    const target = event.target;
    const value = target.value;
    this.setState({
      isError: value !== null && value !== '' && !isUsZip.test(value),
    });
  }

  renderError(): JSX.Element {
    const { errorMessage } = this.props;
    return errorMessage === undefined ? (
      <React.Fragment />
    ) : (
      <p className="search-input-message--error">{errorMessage}</p>
    );
  }

  render(): JSX.Element {
    const { name, pattern, placeholder } = this.props;
    const { isError } = this.state;
    return (
      <React.Fragment>
        {this.renderError()}
        <input
          type="text"
          name={name}
          className={isError ? 'error' : ''}
          pattern={pattern}
          placeholder={placeholder}
          onChange={this.onChange.bind(this)}
        />
      </React.Fragment>
    );
  }
}
