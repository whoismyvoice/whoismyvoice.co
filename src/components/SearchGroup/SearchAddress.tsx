import React from 'react';
import cx from 'classnames';

// Component
import { SearchInput } from './SearchInput';
import TextFormButton from '../Buttons/TextFormButton';

interface Props {
  address?: string;
  addressErrorMessage?: string;
  isStreetAddressNeeded: boolean;
  onSubmit?: (event: React.FormEvent<HTMLFormElement>) => void;
  placeholder?: string;
  zipCode?: string;
}

export class SearchAddress extends React.Component<Props> {
  static defaultProps: Props = {
    address: '',
    addressErrorMessage: undefined,
    isStreetAddressNeeded: false,
    onSubmit: () => undefined,
    placeholder: 'Enter Your Street Address',
    zipCode: '',
  };

  render(): JSX.Element {
    const { address, addressErrorMessage, onSubmit, placeholder } = this.props;
    const inputClasses = cx('input', {
      // eslint-disable-next-line @typescript-eslint/camelcase
      address_search: this.props.isStreetAddressNeeded,
    });
    let formFields: React.ReactFragment = <React.Fragment />;
    if (this.props.isStreetAddressNeeded) {
      const zipCode = this.props.zipCode;
      formFields = (
        <React.Fragment>
          <p>Multiple Congressional Representatives in</p>
          <div key="zipCodeDisplay" className="locked__zip">
            ZIP: {zipCode}
          </div>
          <input key="zipCode" type="hidden" name="zipCode" value={zipCode} />
          <input
            key="address"
            className={inputClasses}
            type="text"
            name="address"
            defaultValue={address}
            placeholder={placeholder}
          />
        </React.Fragment>
      );
    } else {
      formFields = (
        <React.Fragment>
          <SearchInput errorMessage={addressErrorMessage} name="zipCode" />
        </React.Fragment>
      );
    }
    return (
      <form method="GET" onSubmit={onSubmit}>
        {formFields}
        <div className="line-seperator" />
        <TextFormButton text="Continue" />
      </form>
    );
  }
}
