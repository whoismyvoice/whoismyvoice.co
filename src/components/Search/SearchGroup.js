import React from 'react';
import SenateStore from '../../stores/SenateStore';
import {Settings} from '../../constants/SenateConstants';

// Components
import BaseComponent from '../BaseComponent';
import SearchInput from './SearchInput';
import SearchAddress from './SearchAddress';

class SearchGroup extends BaseComponent {
  constructor() {
    super();
    this.state = SenateStore.getMember();
  }

  render() {
    const {zip_code, error, did_search, number_representatives, state_full} = this.state;

  	if (number_representatives > 1 && Settings.chamber === 'house' ||Â did_search && number_representatives === undefined && error) {
  		return <div className="animated fadeIn">
  			<div className="locked__zip">ZIP: {zip_code}</div>
  			<SearchAddress
  				error={error}
          search_address={true}
          zip_code={zip_code}
          state_full={state_full}
  			/>
  		</div>;
  	} else {
    	return <div>
      	<SearchInput
        	error={error}
      	/>
    	</div>;
    }
  }
}

export default SearchGroup;
