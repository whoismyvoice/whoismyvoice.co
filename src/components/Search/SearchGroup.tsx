import * as React from 'react';
import { connect } from 'react-redux';

// Components
import { Dispatch } from '../../actions/types';
import { State } from '../../store';
import SearchAddress from './SearchAddress';
import { setAddress, setZipCode } from '../../actions';

type Props = StateProps & DispatchProps;

interface StateProps {
  address?: string;
  addressErrorMessage?: string;
  isStreetAddressNeeded: boolean;
  zipCode?: string;
}

interface DispatchProps {
  onAddressSubmit: (event: React.FormEvent<HTMLFormElement>) => void;
}

export class SearchGroup extends React.Component<Props> {
  static defaultProps = {
    addressErrorMessage: undefined,
    isStreetAddressNeeded: false,
    onAddressSubmit: () => undefined,
  };

  render() {
    const {
      address,
      addressErrorMessage,
      isStreetAddressNeeded,
      onAddressSubmit,
      zipCode,
    } = this.props;
    return (
      <div className="animated fadeIn">
        <SearchAddress
          address={address}
          addressErrorMessage={addressErrorMessage}
          isStreetAddressNeeded={isStreetAddressNeeded}
          onSubmit={onAddressSubmit}
          zipCode={zipCode}
        />
      </div>
    );
  }
}

function mapStateToProps(state: State): StateProps {
  const { address, officials, view } = state;
  const { addressErrorMessage } = view;
  return {
    address: address.street,
    addressErrorMessage,
    isStreetAddressNeeded: officials.ids.length === 2,
    zipCode: address.zipCode,
  };
}

function mapDispatchToProps(dispatch: Dispatch): DispatchProps {
  return {
    onAddressSubmit: (event: React.FormEvent<HTMLFormElement>) => {
      event.preventDefault();
      const target = event.currentTarget;
      const addressNode = target.address;
      const zipCodeNode = target.zipCode;
      const streetAddress = addressNode && addressNode.value;
      const zipCode = zipCodeNode && zipCodeNode.value;
      if (streetAddress) {
        const address = `${streetAddress}, ${zipCode}`;
        setAddress(address)(dispatch);
      } else {
        setZipCode(zipCode)(dispatch);
      }
    },
  };
}

export default connect(mapStateToProps, mapDispatchToProps)(SearchGroup);
