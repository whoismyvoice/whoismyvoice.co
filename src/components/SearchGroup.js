import React from 'react';

// Components
import BaseComponent from './BaseComponent';
import SearchInput from './SearchInput';
import SearchAddress from './SearchAddress';

class SearchGroup extends BaseComponent {
  render() {
    const {zip_code, error, repNum} = this.props;

  	if (repNum > 3) {
  		return <div>
  			<i>Zip Code: {zip_code}</i><br />
  			<b>Please provide street name to get more specific result</b>
  			<SearchAddress
  				error={error}
  			/>
  		</div>;
  	} else {
    	return <div>
      	<SearchInput
        	error={error}
      	/>
    	</div>;
    }
  }
};

SearchGroup.propTypes = {
  repNum: React.PropTypes.number,
  zip_code: React.PropTypes.string,
  error: React.PropTypes.bool
};

export default SearchGroup;
