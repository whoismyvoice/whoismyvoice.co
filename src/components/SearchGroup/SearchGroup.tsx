import * as React from 'react';

// Components
import { SearchAddress } from './SearchAddress';

export interface StateProps {
  address?: string;
  addressErrorMessage?: string;
  isStreetAddressNeeded?: boolean;
  zipCode?: string;
}

export interface DispatchProps {
  onAddressSubmit?: (event: React.FormEvent<HTMLFormElement>) => void;
}

export interface Props extends StateProps, DispatchProps {}

export class SearchGroup extends React.Component<Props> {
  static defaultProps: Props = {
    addressErrorMessage: undefined,
    isStreetAddressNeeded: false,
    onAddressSubmit: () => undefined,
  };

  render(): JSX.Element {
    const {
      address,
      addressErrorMessage,
      isStreetAddressNeeded,
      onAddressSubmit,
      zipCode,
    } = this.props;
    return (
      <div className="animated fadeIn" data-testid="search-group">
        <SearchAddress
          address={address}
          addressErrorMessage={addressErrorMessage}
          isStreetAddressNeeded={isStreetAddressNeeded}
          onSubmit={onAddressSubmit}
          zipCode={zipCode}
        />
      </div>
    );
  }
}
