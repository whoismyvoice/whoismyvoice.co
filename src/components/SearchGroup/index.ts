import { connect } from 'react-redux';
import { DispatchProps, StateProps, SearchGroup } from './SearchGroup';
import { Dispatch } from '../../actions/types';
import { State } from '../../store';
import { setAddress, setZipCode } from '../../actions';

function mapStateToProps(state: State): StateProps {
  const { address, officials, view } = state;
  const { addressErrorMessage } = view;
  return {
    address: address.street,
    addressErrorMessage,
    isStreetAddressNeeded:
      officials.ids.length === 2 ||
      addressErrorMessage === 'Failed to parse address',
    zipCode: address.zipCode,
  };
}

function mapDispatchToProps(dispatch: Dispatch): DispatchProps {
  return {
    onAddressSubmit: (event: React.FormEvent<HTMLFormElement>) => {
      event.preventDefault();
      const target = event.currentTarget;
      const addressNode = target.address;
      const zipCodeNode = target.zipCode;
      const streetAddress = addressNode && addressNode.value;
      const zipCode = zipCodeNode && zipCodeNode.value;
      if (streetAddress) {
        const address = `${streetAddress}, ${zipCode}`;
        setAddress(address)(dispatch);
      } else {
        setZipCode(zipCode)(dispatch);
      }
    },
  };
}

const connector = connect(mapStateToProps, mapDispatchToProps);
export default connector(SearchGroup);
