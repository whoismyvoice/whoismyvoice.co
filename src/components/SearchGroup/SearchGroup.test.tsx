import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import { render } from '@testing-library/react';

import { SearchGroup } from './SearchGroup';

it('renders without crashing', () => {
  const { container } = render(<SearchGroup isStreetAddressNeeded={false} />);
  expect(container).toBeInTheDocument();
});

it('renders correctly', () => {
  const { getByTestId } = render(<SearchGroup isStreetAddressNeeded={false} />);
  expect(getByTestId('search-group')).toBeInTheDocument();
  expect(getByTestId('search-address')).toBeInTheDocument();
});
