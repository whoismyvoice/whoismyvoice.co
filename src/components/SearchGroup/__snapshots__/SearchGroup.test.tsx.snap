// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="animated fadeIn"
  data-testid="search-group"
>
  <form
    method="GET"
    onSubmit={[Function]}
  >
    <input
      className=""
      name="zipCode"
      onChange={[Function]}
      pattern="[0-9]*"
      placeholder="Enter Your Zip Code"
      type="text"
    />
    <div
      className="line-seperator"
    />
    <button
      className="text-button"
      onClick={[Function]}
      type="submit"
    >
      Continue
      <span
        className="text-button-border"
      />
    </button>
  </form>
</div>
`;
